File: Algorithms/C++/0005/0005.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        if(s.empty()) {
            return "";
        }
        string res; int maxLen = 0;
        for(int i = 0; i < s.size(); i++) {
            int left = i, right = i;
            while(left > 0 && right < s.size()-1 && s[left-1] == s[right+1]) {
                left--; right++;
            }
            if(right-left+1 > maxLen) {
                maxLen = right - left + 1;
                res = s.substr(left, maxLen);
            }
        }
        for(int i = 0; i < s.size()-1; i++) {
            int left = i, right = i+1;
            if(s[left] != s[right]) {
                continue;
            }
            while(left > 0 && right < s.size()-1 && s[left-1] == s[right+1]) {
                left--; right++;
            }
            if(right-left+1 > maxLen) {
                maxLen = right - left + 1;
                res = s.substr(left, maxLen);
            }
        }
        return res;
    }
};

File: Algorithms/Python/0005/0005.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if s == "":
            return ""
        max_length = 0
        start, end = 0, 0
        for i in range(len(s)):
            left, right = i, i
            while left > 0 and right < len(s) - 1 and s[left - 1] == s[right + 1]:
                left, right = left - 1, right + 1
            if right - left + 1 > max_length:
                max_length = right - left + 1
                start, end = left, right
        for i in range(len(s) - 1):
            left, right = i, i + 1
            if s[left] != s[right]:
                continue
            while left > 0 and right < len(s) - 1 and s[left - 1] == s[right + 1]:
                left, right = left - 1, right + 1
            if right - left + 1 > max_length:
                max_length = right - left + 1
                start, end = left, right
        return s[start: end + 1]
